File: fixtures/sample-repos/webdriverio/support/factories.ts
// Factory: external test data. No hardcoded credentials in test bodies.
export interface User {
  email: string;
  password: string;
  role: 'trader' | 'admin' | 'viewer';
}

export const UserFactory = {
  create(overrides: Partial<User> = {}): User {
    const stamp = Date.now().toString(36);
    return {
      email: `user-${stamp}@example.test`,
      password: `pw-${stamp}`,
      role: 'trader',
      ...overrides,
    };
  },
};

File: fixtures/sample-repos/playwright-dotnet/Support/Factories.cs
namespace Sample.PlaywrightDotnet.Support;

public enum UserRole
{
    Trader,
    Admin,
    Viewer,
}

public record User(string Email, string Password, UserRole Role);

public static class UserFactory
{
    public static User Create(UserRole role = UserRole.Trader)
    {
        var stamp = DateTime.UtcNow.Ticks.ToString("x");
        return new User($"user-{stamp}@example.test", $"pw-{stamp}", role);
    }
}
